Plugin/ProductEavDataProviderPlugin.php

```diff
@@ -0,0 +1,85 @@
+<?php
+namespace AvS\ScopeHint\Plugin;
+
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Catalog\Model\Attribute\ScopeOverriddenValue;
+
+class ProductEavDataProviderPlugin
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
```

Comment:
Is there a reason why you are using protected instead of private variables here?